pkg/builder/client/client.go

```diff
@@ -39,12 +38,13 @@ type (
 	Client struct {
 		logger     *zap.Logger
 		url        string
-		httpClient *http.Client
+		httpClient *retryablehttp.Client
 	}
 )
 
 func MakeClient(logger *zap.Logger, builderUrl string) *Client {
-	hc := &http.Client{Transport: otelhttp.NewTransport(http.DefaultTransport)}
+	hc := retryablehttp.NewClient()
```

Comment:
Default max retry time is 30 seconds where as min is 1 in library.
Max tries default is 4.
https://github.com/hashicorp/go-retryablehttp/blob/7f638666931a119b8d7a1239e44da11db0d446a4/client.go#L47

We need to see if we are honoring current max retry time.

Reply:
> However 20 retries is too much we and reduce retries and increase wait duration.

Yes, I agree. The original intention of this PR is to change the number of retries and retry strategy.

The current retry policy is from the PR Istio integration (https://github.com/fission/fission/pull/421), I think it wants to retry only for the network error, not the build error.  short time with 20 retries for the network error may be suitable, But not suitable for the build error.

Reply:
Make sense. As per current logic written in code, max retry time won't not more than 2 second.
So i think we can go ahead with these changes without specifying `hc.RetryMax = 20`